src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -1355,6 +1355,16 @@ TEST_CASE("LedgerTxn load", "[ledgertxn]")
                         NonSociRelatedException);
                 }
 
+                SECTION("invalid asset")
+                {
+                    std::string assetCode;
+                    assetCode.push_back('\n');
+                    auto asset = txtest::makeAsset(acc, assetCode);
+                    REQUIRE_THROWS_AS(
+                        ltx1.load(trustlineKey(acc.getPublicKey(), asset)),
+                        NonSociRelatedException);
+                }
```

Comment:
Can you also test one or two other ways that assets can be invalid? Eg. asset code has non-trailing zeros, and `assetCode12` has fewer than 5 characters. I feel very confident that they will pass, but I still think it's important to test these things.

Reply:
Added. I also added similar tests for revoke sponsorship in https://github.com/stellar/stellar-core/pull/2768.